cluster/juju/layers/kubernetes-worker/reactive/kubernetes_worker.py

```diff
@@ -346,9 +346,12 @@ def start_worker(kube_api, kube_control, auth_control, cni):
     # the correct DNS even though the server isn't ready yet.
 
     dns = kube_control.get_dns()
+    if dns['enable-kube-dns'] == None:
+        hookenv.log('Waiting for dns info.')
+        return
```

Comment:
Can we put this check in interface-kube-control's `KubeControlRequireer.dns_ready` instead? https://github.com/juju-solutions/interface-kube-control/blob/fc7be6163e2eeea71cbec3cefba9ddbd17d26d35/requires.py#L89

Doing so would prevent `kube-control.dns.available` from being set until this condition is met, which I believe is the intended use of that flag.

Reply:
Yes here it is:
https://github.com/juju-solutions/interface-kube-control/pull/18